### PR TITLE
gitlab-runner: update to 15.2.1

### PR DESCRIPTION
**Pull request by herbygillot**
(PR building from source.)

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/gitlab-runner/Portfile [hunk 1]
```diff
@@ -3,41 +3,45 @@
...
-                    size    9144873
+# Allow Go to fetch deps at build time
+build.env-delete    GO111MODULE=off GOPROXY=off
 
 # Reproduce the "build_simple" target from the upstream Makefile
-set go_ldflags      "-X ${go.package}/common.NAME=${go.package} \
+set go_ldflags     " \
+    -X ${go.package}/common.NAME=${go.package} \
     -X ${go.package}/common.VERSION=${version} \
     -X ${go.package}/common.REVISION=unknown \
```
[the remark this answers is not in the document]
Comment (tsabirgaliev) [reply devel/gitlab-runner/Portfile:36]:
```suggestion
    -X ${go.package}/common.REVISION=32fc1585 \
```

File: devel/gitlab-runner/Portfile [hunk 2]
```diff
@@ -3,41 +3,45 @@
...
     -X ${go.package}/common.BUILT=unknown \
     -X ${go.package}/common.BRANCH=unknown \
```
Comment (tsabirgaliev) [inline devel/gitlab-runner/Portfile:38]:
```suggestion
    -X ${go.package}/common.BRANCH=15-2-stable \
```

**Review by tsabirgaliev — CHANGES_REQUESTED** [review]

File: devel/gitlab-runner/Portfile [hunk 3]
```diff
@@ -3,41 +3,45 @@
...
+                    sha256  8c183a13ac6398f9746b2d5d5f0b7640640516ad241286765477e2e85998d5c5 \
+                    size    1119870
 
-checksums           rmd160  f362a2eed88aa25bfaf15ffc6d378610c02838e9 \
-                    sha256  2700a09e1bb468cee8abc01998f7cd99c51f9218e3d0f5d3dcbf334322669869 \
-                    size    9144873
+# Allow Go to fetch deps at build time
+build.env-delete    GO111MODULE=off GOPROXY=off
 
 # Reproduce the "build_simple" target from the upstream Makefile
```
Comment (tsabirgaliev) [inline devel/gitlab-runner/Portfile:32]:
```suggestion
# Reproduce the "build_simple" target from the upstream Makefile
# Lookup REVISION from https://gitlab.com/gitlab-org/gitlab-runner/-/releases/v${version}
```

**Comment by tsabirgaliev** [comment]
@herbygillot now it works

**Comment by herbygillot** [comment]
> @herbygillot now it works

OK, thanks for testing @tsabirgaliev. Checking this in.

**Pull request merged**